numba/datamodel/models.py

```diff
@@ -370,6 +370,53 @@ def load_from_data_pointer(self, builder, ptr, align=None):
         return builder.bitcast(ptr, self.get_value_type())
 
 
+@register_default(types.CVoid)
+class CVoidModel(PrimitiveModel):
+    def __init__(self, dmm, fe_type):
+        be_type = ir.VoidType()
+        super(CVoidModel, self).__init__(dmm, fe_type, be_type)
+
+
+@register_default(types.Vector)
+class VectorModel(PrimitiveModel):
+
+    def __init__(self, dmm, fe_type):
+        dtype_model = dmm.lookup(fe_type.dtype)
+        count = fe_type.count
+        be_type = VectorType(count, dtype_model.get_value_type())
+        super(VectorModel, self).__init__(dmm, fe_type, be_type)
+
+
+#  XXX: this should be part of llvmlite
+class VectorType(ir.Type):
```

Comment:
There are now 2 PR about VectorType in llvmlite.  Numba team needs to resolve them: 
* https://github.com/numba/llvmlite/pull/313
* https://github.com/numba/llvmlite/pull/322